src/scripts/main.js

```diff
@@ -1,11 +1,28 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  // eslint-disable-next-line no-shadow
+
+  return new Promise((resolve) => {
+    // eslint-disable-next-line no-shadow
+    element.addEventListener(eventName, function (event) {
+      event.preventDefault();
```

Comment:
The `event.preventDefault()` method is used here, which is typically necessary for form submissions or anchor tags to prevent default behavior. Ensure that this is needed for the events you are handling (e.g., 'click', 'input', 'blur'). If not, it might be unnecessary.